docs/proposals/proposal-trigger-and-deploy.md

```diff
@@ -0,0 +1,286 @@
+      - ebpf
+      - modern-ebpf
+      - kmod
+```
+
+A scheduled GitHub Action will run every 30 minutes and check the Atom feed of
+each project for new releases.
+
+If a new release is detected the action will trigger the pipeline for the new
+release and update the YAML file with the new version. This is to ensure each
+release is only triggered once.
+
+If sub components are specified then the pipeline will be triggered once per
+sub component.
+
+### Trigger
+
+The green reviews pipeline will be triggered by sending a [workflow_dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)
+event via its GitHub webhook.
+
+Inputs are
+
+- `cncf_project`: **required** Project to be deployed e.g. `falco`
+- `cncf_project_sub`: **optional** Subcomponent if project has multiple variants
+they wish to test e.g. `modern-ebpf`
+- `version`: **required** Version of project to be tested e.g. `0.37.0`
+
+```sh
+curl -X POST \
+     -H "Accept: application/vnd.github.v3+json" \
+     -H "Authorization: token $GITHUB_PAT" \
+     https://api.github.com/repos/cncf-tags/green-reviews-tooling/actions/workflows/pipeline.yaml/dispatches \
+     -d '{"ref":"main", "inputs": {"cncf_project": "falco", "cncf_project_sub": "modern-ebpf","version":"0.37.0"}}'
+```
+
+The CNCF projects will be given a GitHub fine grained access token limited to
+the `green-reviews-tooling` repo. This token will have
+
+- Read access to code and metadata
+- Read write access to actions
+
+### Deploy
+
+Flux will be used to deploy the CNCF project. Projects are able to use either
+`kustomization` or `helmrelease` resources to deploy their project.
+
+Project resources that should always be deployed in the cluster are stored in
+the current location in the tooling repo, below are format where we might store
+project related configurations
+```
+# for the cncf_project
+clusters/projects/${project_name}
+
+# and for each cncf_project (different configurations)
+clusters/projects/${project_name}/${configuration_name}
+```
+
+e.g. `/clusters/projects/falco/`
+and are reconciled by source-controller.
+
+When the pipeline executes it will look for yaml files in the projects dir.
+If there is a yaml file matching the `cncf_project` input its contents will be
+applied using kubectl. The same applies for the `cncf_project_sub` input. 
+
+The `version` param will need to be injected into the files to ensure the
+correct version of the project is deployed. 
+(For these small minor changes we can utilize kustomize)
+
+```
+projects
+└── falco
+    ├── ebpf.yaml
+    ├── falco.yaml
+    ├── kmod.yaml
+    └── modern-ebpf.yaml
+```
+
+The pipeline will use a GitHub secret that has a kubeconfig to access the
+green reviews cluster. The deploy step in the pipeline will wait for the newly
+created flux resources to be reconciled before proceeding to the run step.
+
+We will have a node to deploy the project and another to run the benchmarks
```

Comment:
Do you mean this? Otherwise I don't understand why node is mentioned here.
```suggestion
We will have a job to deploy the project and another to run the benchmarks
```

Reply:
The cluster currently has 2 worker nodes. One node is used to deploy falco and the other node has the rest of the components including the benchmarks, prometheus, grafana etc.

This is to mitigate noisy neighbour type problems affecting the energy measurements of falco. Same here I'll rework this section to be clearer.